promql/test.go

```diff
@@ -51,33 +52,50 @@ const (
 
 var testStartTime = time.Unix(0, 0).UTC()
 
+//go:embed testdata
+var TestManifestFS embed.FS // TestManifestFS contains a set of tests.
+
+// TestEngine is the interface for an engine under test.
+type TestEngine interface {
+	NewRangeQuery(q storage.Queryable, opts *QueryOpts, qs string, start, end time.Time, interval time.Duration) (Query, error)
+	NewInstantQuery(q storage.Queryable, opts *QueryOpts, qs string, ts time.Time) (Query, error)
+}
+
 // Test is a sequence of read and write commands that are run
 // against a test storage.
-type Test struct {
+type Test[T TestEngine] struct {
```

Comment:
Had to use generics now so that `suite.QueryEngine()` doesnt need type assertions; this is a mildly breaking change sadly.